File: sem 4/JIPP/lab2/zad2/ModuloFieldTest.java
package zad2;

public class ModuloFieldTest {
    public static void main(String[] args) {
        try {
            // Larger numbers for demonstration
            ModuloField a = new ModuloField(123456);
            ModuloField b = new ModuloField(987654);
            ModuloField c = new ModuloField(654321);

            System.out.println("Value of a: " + a.getValue());
            System.out.println("Value of b: " + b.getValue());
            System.out.println("Value of c: " + c.getValue());

            // Testing mathematical operations
            ModuloField resultAdd = ModuloField.add(a, b);
            assert resultAdd != null && resultAdd.getValue() == 1111110;
            System.out.println("\nAddition:");
            System.out.println(a.getValue() + " + " + b.getValue() + " = " + resultAdd.getValue());

            ModuloField resultSub = ModuloField.substract(c, b);
            assert resultSub != null && resultSub.getValue() == 901244;
            System.out.println("\nSubtraction:");
            System.out.println(c.getValue() + " - " + b.getValue() + " = " + resultSub.getValue());

            ModuloField resultMul = ModuloField.multiply(a, c);
            assert resultMul != null && resultMul.getValue() == 245689;
            System.out.println("\nMultiplication:");
            System.out.println(a.getValue() + " * " + c.getValue() + " = " + resultMul.getValue());

            ModuloField resultDiv = ModuloField.divide(c, b);
            assert resultDiv != null && resultDiv.getValue() == 1180971;
            System.out.println("\nDivision:");
            System.out.println(c.getValue() + " / " + b.getValue() + " = " + resultDiv.getValue());

            // Testing equality and comparisons
            System.out.println("\nEquality Test:");
            assert a.equalTo(a);
            assert !a.equalTo(b);

            System.out.println("\nComparison Test:");
            assert a.lessThan(b);
            assert !a.greaterThan(c);
            assert a.lessOrEqual(b);
            assert !a.greaterOrEqual(c);

            System.out.println("All assertions passed.");
        } catch (CharacteristicException e) {
            System.out.println("CharacteristicException: " + e.getMessage());
        }
    }
}

File: sem 4/JIPP/lab2/zad3/ModuloFieldTest.py
from ModuloField import ModuloField, CharacteristicException

def main():
    try:
        # Larger numbers for demonstration
        a = ModuloField(123456)
        b = ModuloField(987654)
        c = ModuloField(654321)

        print("Value of a:", a.get_value())
        print("Value of b:", b.get_value())
        print("Value of c:", c.get_value())

        # Testing mathematical operations
        result_add = a + b
        assert result_add.get_value() == 1111110
        print("\nAddition:")
        print(f"{a.get_value()} + {b.get_value()} = {result_add.get_value()}")

        result_sub = c - b
        assert result_sub.get_value() == 901244
        print("\nSubtraction:")
        print(f"{c.get_value()} - {b.get_value()} = {result_sub.get_value()}")

        result_mul = a * c
        assert result_mul.get_value() == 245689
        print("\nMultiplication:")
        print(f"{a.get_value()} * {c.get_value()} = {result_mul.get_value()}")

        result_div = c / b
        assert result_div.get_value() == 1180971
        print("\nDivision:")
        print(f"{c.get_value()} / {b.get_value()} = {result_div.get_value()}")

        # Testing equality and comparisons
        print("\nEquality Test:")
        assert a == a
        assert not (a == b)

        print("\nComparison Test:")
        assert a < b
        assert not (a > c)
        assert a <= b
        assert not (a >= c)

        print("All assertions passed.")

    except CharacteristicException as e:
        print("CharacteristicException:", e)

if __name__ == "__main__":
    main()
